app/controllers/api/v1/reset_password_controller.rb

```diff
@@ -23,6 +24,46 @@ def create
 
         render_json data: { token: } # TODO: enable reset email sending.
       end
+
+      # POST /api/v1/reset_password/reset.json
+      # Expects: { user: {:token, :new_password} }
+      # Returns: { data: Array[serializable objects] , errors: Array[String] }
+      # Does: Validates the token and reset the user password.
+
+      def reset
+        new_password = params[:user][:new_password]
+
+        return render_json status: :bad_request if new_password.blank?
+
+        # Invalidating token and changing the password.
+        # TODO: optimise this.
+        @user.update! password: new_password
+        @user.invalidate_reset_token!
+
+        render_json
+      end
+
+      # POST /api/v1/reset_password/verify.json
+      # Expects: { user: {:token} }
+      # Returns: { data: Array[serializable objects] , errors: Array[String] }
+      # Does: Validates the token.
+
+      def verify
+        render_json
+      end
+
+      private
+
+      def verify_reset_request
+        return render_json status: :bad_request unless params[:user]
+
+        token = params[:user][:token]
+
+        return render_json status: :bad_request if token.blank?
+
+        @user = User.verify_reset_token!(token)
+        render_json status: :forbidden unless @user
+      end
```

Comment:
I think that using unless and if's in the same block, back to back, makes it hard to follow (maybe it's just me).

For example, I would do `if params[:user].empty? `on line 58

Also, shouldn't we use validations for that kind of thing? If it's missing a param or if it's blank?
Or have another solution? It clutters the controller IMO.

Reply:
Do you like using unless instead of if statements?

As for the missing it's intentional for security reasons to obscure the responses to avoid given information about sensitive operations such as the reset password.
For example if we return an error status code for an invalid token we're giving attackers a potential attack vector that can find a token in use in our system.
Another example would be on generating the token for a user that is found by email, returning an error like `:not_found` will put an endpoint to guess our users emails from the outside.

Reply:
I think I misunderstood you, you meant we should be using the `ActionController::Parameters` API for params validation?

Reply:
Do you like using `unless` instead of if statements?
As for the validations are you suggesting the use of the `ActionController::Parameters` API for params validation?